pymodm/fields.py

```diff
@@ -1180,15 +1199,23 @@ def contribute_to_class(self, cls, name):
                                                     self._on_delete)
 
     def to_python(self, value):
+        doc = value
         if isinstance(value, dict):
             # Try to convert the value into our document type.
-            return self.related_model.from_document(value)
-        elif isinstance(value, self.related_model):
-            return value
+            try:
+                doc = self.related_model.from_document(value)
+            except Exception:
```

Comment:
I think only `ValueError` is possible here, from this line: https://github.com/mongodb/pymodm/blob/ce603f5079fb602eeb2421e1bca767f4cb2bd8ac/pymodm/base/models.py#L248. No `ValidationError`s should be triggered by this. Are there other cases I'm not thinking of?